source/dev/idea_list.md

```diff
@@ -0,0 +1,148 @@
+
+Note that this project will eventually incorporate the work of the Simulation Data Validation Manager and Simulation Data Download Manager projects. However, each of these projects can proceed in parallel.
+
+
+- Write Apache Airflow workflows that periodically 
+    - Query external data sources for availability
+    - Download external data sources
+    - Validate and clean the downloaded resources
+    - Archive downloaded resources to a blob store
+- Workflows are instrumented with telemetry
+- Telemetry are aggregated in InfluxDB
+- Metrics are displayed in a Grafana dashboard
+- If a data source is down, we throw an alarm
+The workflows should be well documented with a focus on code quality, and using best-of-breed current technology stacks (Apache Airflow, Kubernetes, Helm, Docker, Python3, Grafana, etc.)
+
+**Requirements:**
+Docker, Python, Grafana, PAAS, Apache Airflow
+
+**Level:** 
+Medium to Hard
+
+**Mentor:**
+
+### Project 4. - Test system for end-to-end testing of the simulation framework 
+Use [existing test power systems](http://labs.ece.uw.edu/pstca/) that will be used in our simulation framework. 
+Generate input data to feed to the simulation engine using data science techniques.
+Write integration tests and produce documentation to showcase our open source software.
+
+- Collect data (python scripting)
+- Generate time series for demand, hydro, solar and wind matching the test grid (python data analysis)
+- Integrate the test model in our [simulation framework](https://github.com/Breakthrough-Energy/PowerSimData) (python)
+- Write end-to-end tests (in docker container)
+- Write GitHub actions to test changes of the framework when pushed on GitHub
+- Write documentation for the community (sphinx)
+
+**Requirements:**
+Object Oriented Programming in Python, Data analysis/plotting in Python (pandas, numpy, matplotlib, etc.) and docker
+
+**Level:** 
+Easy
+
+**Mentor:**
+[Ben](https://github.com/rouille)
+
+### Project 5. - Time-Series Modeling of Missing Energy Industry Data
+A significant barrier to realistic modeling of the power grid is missing data from public data sources.
+This project is to use statistical and ML models to fill in missing energy demand and renewable energy profile information in a way which mimics the real data as close as possible.
+In our current datasets of regional energy demand, we have used averaging of adjacent days to fill in our missing data, as described [here](https://github.com/Breakthrough-Energy/PreREISE#d-demand-data)
+
+We propose the following steps for this project:
+
+
+- Understand the data set and the [current methodology](https://github.com/Breakthrough-Energy/PreREISE#d-demand-data)
+- Literature review to understand prediction models for time series data
+- Collect data to build training/validation/testing sets to rank algorithms using various metrics
+- Setup platform to crossvalidate with the data in order to evaluate algorithm performance
```

Comment:
I'd change "setup" to "set up" (the former is usually a noun). Actually, I might reorganize these two points slightly. Would have "collect data to build training/testing sets" as one point, then "evaluate algorithms using cross validation or other metrics" as the next one. Not sure it's necessary to be a "platform"

Reply:
I think 'crossvalidate' should probably be two words: 

https://en.wikipedia.org/wiki/Cross-validation_(statistics)

Reply:
@rouille, same here.

Reply:
I like what @jon-hagg  proposed